tests/components/influxdb/test_sensor.py

```diff
@@ -0,0 +1,453 @@
+)
+async def test_full_config(hass, mock_client, config_ext, queries):
+    """Test the full config."""
+    await _setup(hass, config_ext, queries, ["sensor.test"])
+
+
+@pytest.mark.parametrize("config_ext", [(BASE_V1_CONFIG), (BASE_V2_CONFIG)])
+async def test_config_failure(hass, config_ext):
+    """Test an invalid config."""
+    config = {"platform": DOMAIN}
+    config.update(config_ext)
+
+    with pytest.raises(Invalid):
+        PLATFORM_SCHEMA(config)
+
+
+@pytest.mark.parametrize(
+    "mock_client, config_ext, queries, set_query_mock, make_resultset",
+    [
+        (
+            DEFAULT_API_VERSION,
+            BASE_V1_CONFIG,
+            BASE_V1_QUERY,
+            _set_query_mock_v1,
+            _make_v1_resultset,
+        ),
+        (
+            API_VERSION_2,
+            BASE_V2_CONFIG,
+            BASE_V2_QUERY,
+            _set_query_mock_v2,
+            _make_v2_resultset,
+        ),
+    ],
+    indirect=["mock_client"],
+)
+async def test_state_matches_query_result(
+    hass, mock_client, config_ext, queries, set_query_mock, make_resultset
+):
+    """Test state of sensor matches respone from query api."""
+    set_query_mock(mock_client, return_value=make_resultset(42))
+
+    sensors = await _setup(hass, config_ext, queries, ["sensor.test"])
+
+    assert sensors[0].state == "42"
+
+
+@pytest.mark.parametrize(
+    "mock_client, config_ext, queries, set_query_mock, make_resultset",
+    [
+        (
+            DEFAULT_API_VERSION,
+            BASE_V1_CONFIG,
+            BASE_V1_QUERY,
+            _set_query_mock_v1,
+            _make_v1_resultset,
+        ),
+        (
+            API_VERSION_2,
+            BASE_V2_CONFIG,
+            BASE_V2_QUERY,
+            _set_query_mock_v2,
+            _make_v2_resultset,
+        ),
+    ],
+    indirect=["mock_client"],
+)
+async def test_state_matches_first_query_result_for_multiple_return(
+    hass, caplog, mock_client, config_ext, queries, set_query_mock, make_resultset
+):
+    """Test state of sensor matches respone from query api."""
+    set_query_mock(mock_client, return_value=make_resultset(42, "not used"))
+
+    sensors = await _setup(hass, config_ext, queries, ["sensor.test"])
+    assert sensors[0].state == "42"
+    assert (
+        len([record for record in caplog.records if record.levelname == "WARNING"]) == 1
+    )
+
+
+@pytest.mark.parametrize(
+    "mock_client, config_ext, queries, set_query_mock",
+    [
+        (DEFAULT_API_VERSION, BASE_V1_CONFIG, BASE_V1_QUERY, _set_query_mock_v1,),
+        (API_VERSION_2, BASE_V2_CONFIG, BASE_V2_QUERY, _set_query_mock_v2),
+    ],
+    indirect=["mock_client"],
+)
+async def test_state_for_no_results(
+    hass, caplog, mock_client, config_ext, queries, set_query_mock
+):
+    """Test state of sensor matches respone from query api."""
+    set_query_mock(mock_client)
+
+    sensors = await _setup(hass, config_ext, queries, ["sensor.test"])
+    assert sensors[0].state == STATE_UNKNOWN
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5760157)